cookbooks/cdo-varnish/libraries/http_cache.rb

```diff
@@ -69,12 +69,21 @@ def self.config(env)
             proxy: 'dashboard',
             headers: LANGUAGE_HEADER,
             cookies: whitelisted_cookies
+          },
+          {
+            path: %w(
+              /
+              /learn*
+              /congrats
+            ),
+            headers: LANGUAGE_HEADER,
+            cookies: LANGUAGE_COOKIE
           }
         ],
-        # Default Pegasus paths are cached but language-specific, whitelist only language cookie/header.
+        # Remaining Pegasus paths are English-only and don't require any extra headers or cookies.
```

Comment:
I only vaguely remembered the extra non-english-only whitelist myself until I investigated the root cause of the language bug the other day, and this refactoring is my attempt to get rid of both the bug and the extra whitelist for good.

The whole point and main benefit of this PR that it removes the separate whitelist by implementing the remaining needed features in Varnish (that CloudFront already supports) - with this, there is nothing more to document or explain beyond the existing cache-config.